File: leetcode solutions/C++/valid anagram/validAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        int n = s.length();
        unordered_map<char, int> countMap;

        for (int i =0; i < n; i++) {
            countMap[s[i]]++;
            countMap[t[i]]--;
        }

        for (auto count : countMap)
            if (count.second) return false;
        return true;
    }
};



// sorting
class Solution {
public:
    bool isAnagram(string s, string t) {
        sort(s.begin(), s.end());
        sort(t.begin(), s.end());
        return s == t;
    }
};

File: leetcode solutions/Python/valid anagram/validAnagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        countMap = {}

        for ch in s:
            if ch.lower() in countMap:
                countMap[ch.lower()] += 1
            else:
                countMap[ch.lower()] = 1

        for ch in t:
            if ch.lower() in countMap:
                countMap[ch.lower()] -= 1
            else:
                countMap[ch.lower()] = -1

            if countMap[ch.lower()] < 0:
                return False


        return True

